zanata-war/src/main/java/org/zanata/action/VersionGroupJoinAction.java

```diff
@@ -149,14 +152,45 @@ public String send() {
         }
     }
 
-    @AllArgsConstructor
-    public final class SelectableProject {
+    /**
+     * This is to reset data when user closes dialog or after sending email.
+     * See version-group/request_join_modal.xhtml#cancelJoinGroupEmail
+     */
+    public void resetData() {
+        projectSlug = "";
+        projectVersions.clear();
+        setQuery("");
+    }
 
-        @Getter
-        private HProjectIteration projectIteration;
+    // clear all the project versions in the selected project
+    private void clearProjectVersions() {
+        if(!projectVersions.isEmpty()) {
+            projectVersions.clear();
+        }
+    }
+
+    @Override
+    public List<HProject> suggest() {
+        clearProjectVersions();
+        return projectDAO.getProjectsForMaintainer(
+                authenticatedAccount.getPerson(), getQuery(), 0,
+                Integer.MAX_VALUE);
```

Comment:
We need to clear the all the versions displayed in dialog from previous selected project when user is entering a new project search(autocomplete)